src/clusterfuzz/_internal/bot/tasks/task_types.py

```diff
@@ -24,6 +27,30 @@
 from clusterfuzz._internal.system import environment
 
 
+def _emit_trusted_task_event(task_command,
+                             event_data,
+                             task_status,
+                             task_outcome=None):
+  """Helper to emit task execution events for trusted tasks."""
+
+  if task_utils.is_fuzzer_based_task(task_command):
+    task_event = events.FuzzerTaskExecutionEvent(
+        **event_data,
+        task_stage=events.TaskStage.NA,
+        task_status=task_status,
+        task_outcome=task_outcome,
+        task_comments=environment.get_value('TASK_COMMENTS'))
+  else:
+    task_event = events.TaskExecutionEvent(
+        **event_data,
+        task_stage=events.TaskStage.NA,
+        task_status=task_status,
+        task_outcome=task_outcome,
+        task_comments=environment.get_value('TASK_COMMENTS'))
+
+  events.emit(task_event)
```

Comment:
You can do something like
```
  event_class = (events.FuzzerTaskExecutionEvent
                 if task_utils.is_fuzzer_based_task(task_command)
                 else events.TaskExecutionEvent)

  task_event = event_class(
      **event_data,
      task_stage=events.TaskStage.NA,
      task_status=task_status,
      task_outcome=task_outcome,
      task_comments=environment.get_value('TASK_COMMENTS'))

  events.emit(task_event)
```
to avoid the duplication of the params, but this is a nit. I'll let you decide.